bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -2077,27 +1703,24 @@ else if (i.alphaData != null) {
 			if (hIcon == 0) SWT.error(SWT.ERROR_NO_HANDLES);
 			OS.DeleteObject(hBitmap);
 			OS.DeleteObject(hMask);
-			image.setHandleForZoomLevel(hIcon, zoom);
+			image.new ImageHandle(hIcon, zoom);
 			image.type = SWT.ICON;
 		}
 	} else {
 		if (image == null) {
 			result = new long []{hDib};
 		} else {
-			image.setHandleForZoomLevel(hDib, zoom);
+			image.new ImageHandle(hDib, zoom);
 			image.type = SWT.BITMAP;
 			image.transparentPixel = i.transparentPixel;
 		}
 	}
 	return result;
 }
 
-private void setHandleForZoomLevel(long handle, Integer zoom) {
-	if (this.handle == 0) {
-		this.handle = handle;	// Set handle for default zoom level
-	}
-	if (zoom != null && !zoomLevelToHandle.containsKey(zoom)) {
-		zoomLevelToHandle.put(zoom, handle);
+private void setImageMetadataForHandle(ImageHandle imageMetadata, Integer zoom) {
+	if (zoom != null && !zoomLevelToImageHandle.containsKey(zoom)) {
```

Comment:
Wouldn't it be a proper response to throw an exception if someone tries to override the existing metadata for a given `zoom` ? This way you are silently ignoring the case, which may have unintended consequences in the future.